packages/salesforcedx-vscode-apex/src/languageServerOrphanHandler.ts

```diff
@@ -0,0 +1,116 @@
+
+const ADVICE = nls.localize('orphan_process_advice');
+const YES = nls.localize('yes');
+const CANCEL = nls.localize('cancel');
+const SHOW_PROCESSES = nls.localize('terminate_show_processes');
+const TERMINATE_PROCESSES_BTN = nls.localize('terminate_processes');
+const SHOW_PROCESSES_BTN = nls.localize('terminate_show_processes');
+const DISMISSED_DEFAULT = nls.localize('dismissed');
+const PROCESS_ID = nls.localize('process_id');
+const PROCESS_PARENT_ID = nls.localize('parent_process_id');
+const COMMAND = nls.localize('process_command');
+
+// these messages contain replacable parameters, cannot localize yet
+const CONFIRM = 'terminate_processes_confirm';
+const TERMINATE_ORPHANGED_PROCESSES = 'terminate_orphaned_language_server_instances';
+const TERMINATED_PROCESS = 'terminated_orphaned_process';
+const TERMINATE_FAILED = 'terminate_failed';
+
+export async function resolveAnyFoundOrphanLanguageServers(): Promise<void> {
+  const orphanedProcesses = findAndCheckOrphanedProcesses();
+  if (orphanedProcesses.length > 0) {
+    if (await getResolutionForOrphanProcesses(orphanedProcesses)) {
+      telemetryService.sendEventData('apexLSPStartup', undefined, { orphanCount: orphanedProcesses.length, didTerminate: 1 });
+      for (const processInfo of orphanedProcesses) {
+        try {
+          await terminateProcess(processInfo.pid);
+          telemetryService.sendEventData('apexLSPStartup', undefined, { terminateSuccessful: 1 });
+          showProcessTerminated(processInfo);
+        } catch (err) {
+          showTerminationFailed(processInfo, err);
+          telemetryService.sendEventData(
+            'apexLSPStartup',
+            { terminationErrorMessage: typeof err === 'string' ? err : err?.message ? err.message : 'unknown' },
+            { terminateSuccessful: 0 });
+        }
+      }
+    } else {
+      telemetryService.sendEventData('apexLSPStartup', undefined, { orphanCount: orphanedProcesses.length, didTerminate: 0 });
+    }
+  }
+}
+
+/**
+ * Ask the user how to resolve found orphaned language server instances
+ * @param orphanedProcesses
+ * @returns boolean
+ */
+export async function getResolutionForOrphanProcesses(orphanedProcesses: ProcessDetail[]): Promise<boolean> {
+  const orphanedCount = orphanedProcesses.length;
+
+  if (orphanedCount === 0) {
+    return false;
+  }
+
+  let choice: string | undefined = nls.localize(SHOW_PROCESSES);
+  do {
+    choice = await vscode.window.showWarningMessage(
+      nls.localize(
+        TERMINATE_ORPHANGED_PROCESSES,
+        orphanedCount
+      ),
+      TERMINATE_PROCESSES_BTN,
+      SHOW_PROCESSES_BTN
+    ) ?? DISMISSED_DEFAULT;
+
+    if (requestsTermination(choice) && await terminationConfirmation(orphanedCount)) {
+      return true;
+    } else if (showProcesses(choice)) {
+      const processId: string = PROCESS_ID;
+      const parentProcessId: string = PROCESS_PARENT_ID;
+      const processCommand: string = COMMAND;
```

Comment:
In QE [doc](https://salesforce.quip.com/WlLPAzrvFTyU) I found the output here is wrapping strangly in the table.  Can we use the [table](https://github.com/forcedotcom/salesforcedx-vscode/blob/69edd89386d8287230b5230423d9e3b44a32a981/packages/salesforcedx-utils-vscode/src/output/table.ts#L21) module we use for shared lib output for a more consist text output?